ci-operator/step-registry/openshift-extended/test/openshift-extended-test-commands.sh

```diff
@@ -224,6 +224,28 @@ oc wait nodes --all --for=condition=Ready=true --timeout=15m
 oc wait clusteroperators --all --for=condition=Progressing=false --timeout=15m
 oc get clusterversion version -o yaml || true
 
+function remove_kubeadmin_user() {
+    if [[ "$KUBEADMIN_REMOVED" == "true" ]]; then
+        echo '$KUBEADMIN_REMOVED is set to "true". Deleting kubeadmin secret...'
+        ret_delete_admin=0
+        ## it is mgmt cluster of hypershift or non-hypershift cluster (normal cluster)
+        oc --kubeconfig="${SHARED_DIR}/kubeconfig" delete secrets kubeadmin -n kube-system || ret_delete_admin=$?
```

Comment:
/hold
@dpuniaredhat I see Kui Wang added `return` before the "// it is mgmt cluster of hypershift or non-hypershift cluster (normal cluster)" comment. Is it possible --kubeconfig="${SHARED_DIR}/kubeconfig be a hosted cluster? Need we check it is not a hosted cluster first?

Reply:
please follow https://github.com/openshift/release/pull/50965#discussion_r1568273038

for hosted cluster, there is still `${SHARED_DIR}/kubeconfig` pointed to host cluster (it is copied from ${SHARED_DIR}/nested_kubeconfig). and `${SHARED_DIR}/mgmt_kubeconfig` is pointed to mgmt cluster.
so,
1, need to determine if it is hosted cluster with `if test -f "${SHARED_DIR}/nested_kubeconfig" && diff -q "${SHARED_DIR}/nested_kubeconfig" "${SHARED_DIR}/kubeconfig" >/dev/null;`
2, delete it in namespace of mgmt cluster with `oc --kubeconfig="${SHARED_DIR}/mgmt_kubeconfig" delete secret/kubeadmin-password -n "clusters-${HOSTED_CLUSTER_NAME}" || ret_delete_admin=$?`

Reply:
so,
```shell
function remove_admin_user() {
    if [[ "$KUBEADMIN_REMOVED" == "true" ]]; then
        echo "KUBEADMIN_REMOVED is set to 'true'. Deleting kubeadmin secret..."
        ret_delete_admin=0

        // it is hosted cluster only for workflow cucushift-installer-rehearse-aws-ipi-ovn-hypershift-guest
        // it means you only see hosted cluster in testing
        if test -f "${SHARED_DIR}/nested_kubeconfig" && diff -q "${SHARED_DIR}/nested_kubeconfig" "${SHARED_DIR}/kubeconfig" >/dev/null; then
            echo "it is hosted cluster and no need to delete it from its mgmt cluster."
            return
        fi

        // it is mgmt cluster of hypershift or non-hypershift cluster (normal cluster)
        oc --kubeconfig="${SHARED_DIR}/kubeconfig" delete secrets kubeadmin -n kube-system || ret_delete_admin=$?
        if [ "W${ret_delete_admin}W" != "W0W" ]; then
            echo "fail to delete kubeadmin-password in mgmt clusger or non-hypershift cluster"
            if [ "W${FORCE_SUCCESS_EXIT}W" == "WnoW" ]; then
                echo "do not force success exit"
                exit 1
            fi
            echo "force success exit"
            exit 0
        fi
        echo "Kubeadmin secret deleted successfully for mgmt cluster or non-hypershift cluster."

    else
        echo "KUBEADMIN_REMOVED not set to 'true'. Skipping deletion."
    fi
}
```

Reply:
or, take it
```shell
function remove_admin_user() {
    if [[ "$KUBEADMIN_REMOVED" == "true" ]] && ( [ ! -f "${SHARED_DIR}/nested_kubeconfig" ] || 
        ! diff -q "${SHARED_DIR}/nested_kubeconfig" "${SHARED_DIR}/kubeconfig" >/dev/null); then
        echo "KUBEADMIN_REMOVED is set to 'true' and it is not hosted cluster. Deleting kubeadmin secret..."
        ret_delete_admin=0

        // it is not hosted cluster only for workflow cucushift-installer-rehearse-aws-ipi-ovn-hypershift-guest which does not need to delete it
        // it is mgmt cluster of hypershift or non-hypershift cluster (normal cluster)
        oc --kubeconfig="${SHARED_DIR}/kubeconfig" delete secrets kubeadmin -n kube-system || ret_delete_admin=$?
        if [ "W${ret_delete_admin}W" != "W0W" ]; then
            echo "fail to delete kubeadmin-password in mgmt clusger or non-hypershift cluster"
            if [ "W${FORCE_SUCCESS_EXIT}W" == "WnoW" ]; then
                echo "do not force success exit"
                exit 1
            fi
            echo "force success exit"
            exit 0
        fi
        echo "Kubeadmin secret deleted successfully for mgmt cluster or non-hypershift cluster."

    else
        echo "KUBEADMIN_REMOVED not set to 'true' or it is hosted cluster. Skipping deletion."
    fi
}
```

Reply:
I've updated the comment https://github.com/openshift/release/pull/50965#discussion_r1591925817 and now again running further tests to confirm the condition is working.